src/app/utils/ObjectUtils.ts

```diff
@@ -19,71 +19,80 @@ export class ObjectDifference {
         for (const addedKey of addedKeys) {
             if (after[addedKey] != null) {
                 this.added[addedKey] = after[addedKey];
             }
         }
     }
-    public addCommonKey(commonKeys: string[],
-                        after: Dictionary,
-                        before: Dictionary)
-    : void
-    {
+    private addCommonKeys(commonKeys: string[], after: FirebaseJSONObject, before: FirebaseJSONObject): void {
         for (const commonKey of commonKeys) {
-            if (after[commonKey] == null) {
-                if (before[commonKey] != null) {
-                    this.removed[commonKey] = before[commonKey];
-                }
-            } else if (before[commonKey] == null) {
-                if (after[commonKey] != null) {
-                    this.added[commonKey] = after[commonKey];
-                }
-            } else if (['function', 'symbol', 'bigint'].includes(typeof before[commonKey])) {
-                throw new Error('Not implemented yet');
-            } else if (typeof before[commonKey] === 'undefined') {
-                throw new Error('YOU RE NOT A REAL VALUE ' + commonKey);
-            } else if (['string', 'boolean', 'number'].includes(typeof before[commonKey])) {
-                if (before[commonKey] !== after[commonKey]) {
-                    this.modified[commonKey] = after[commonKey];
-                }
-            } else if (typeof before[commonKey]['length'] === 'number') { // LIST
-                if (typeof after[commonKey]['length'] === 'number') {
-                    if (before[commonKey].length === after[commonKey].length) {
-                        let equal: boolean = true;
-                        for (let i: number = 0; equal && i < before[commonKey]['length']; i++) {
-                            const elementDiff: ObjectDifference = ObjectDifference.from(before[commonKey][i],
-                                                                                        after[commonKey][i]);
-                            const nbDiff: number = elementDiff.countChanges();
-                            if (nbDiff > 0) {
+            this.addCommonKey(commonKey, before[commonKey], after[commonKey]);
+        }
+    }
+    private addCommonKey(commonKey: string, before: FirebaseJSONValue, after: FirebaseJSONValue): void {
+        if (after == null) {
+            if (before != null) {
+                this.removed[commonKey] = before;
+            }
+        } else if (before == null) {
+            // We know after != null at this point (otherwise we'd have taken the previous branch)
+            this.added[commonKey] = after;
+        } else if (['string', 'boolean', 'number'].includes(typeof before)) {
+            if (before !== after) {
+                this.modified[commonKey] = after;
+            }
+        } else if (Array.isArray(before)) {
+            if (Array.isArray(after)) {
+                if (before.length === after.length) {
```

Comment:
tu peux mettre ça dans une fonction this.addCommonListKey ?

Reply:
bueno, clarification. Fait